### PR TITLE
Modify and add negative tests

### PR DESCRIPTION
**Pull request by danny4478**
This change is [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/44)

### CODE REVIEW

File: spm/spm_server.c [hunk 1]
```diff
@@ -507,9 +507,15 @@ void psa_notify(int32_t partition_id)
 
 void psa_clear(void)
 {
-    int32_t flags = (int32_t)osThreadFlagsClear(PSA_DOORBELL);
-    SPM_ASSERT(flags >= 0);
-    PSA_UNUSED(flags);
+    uint32_t flags = osThreadFlagsClear(PSA_DOORBELL);
+
+    // osThreadFlagsClear() asserts the msb on failure
+    SPM_ASSERT(!(flags & 0x80000000));
```
Comment (orenc17) [inline spm/spm_server.c]: i think this should be panic
and please write it like this `((flags & 0x80000000) == 0x80000000)`
and maybe this change should also be made for `psa_eoi()`

File: TESTS/spm/neg_server_tests/neg_ipc_tests.cpp [hunk 2]
```diff
@@ -312,12 +346,33 @@ void server_write_on_wraparound_msg_ptr()
...
     psa_call(handle, NULL, 0, NULL, 0);
 
-    TEST_FAIL_MESSAGE("server_write_from_excese_outvec negative test failed at client side");
+    TEST_FAIL_MESSAGE("server_write_from_excess_outvec negative test failed at client side");
+}
+
+void server_write_with_size_overflow()
+{
+    psa_handle_t handle = 0;
+
```
Comment (orenc17) [inline TESTS/spm/neg_server_tests/neg_ipc_tests.cpp]: `psa_handle_t handle = negative_server_ipc_tests_connect(PART2_WRITE_SIZE_OVERFLOW, MINOR_VER);`

and remove all the freaking empty lines!!!!!!!!

File: TESTS/spm/neg_server_tests/neg_ipc_tests.cpp [hunk 3]
```diff
@@ -147,6 +148,19 @@ void server_read_null_handle()
...
 }
 
+//Testing server read buffer is null
+void server_read_null_buffer()
+{
+    psa_handle_t handle = 0;
+    uint32_t data = 52;
+    psa_invec_t data_vec = {&data, sizeof(data)};
+    handle = negative_server_ipc_tests_connect(PART2_READ_NULL_BUFFER, MINOR_VER);
+
```
Comment (orenc17) [inline TESTS/spm/neg_server_tests/neg_ipc_tests.cpp]: remove empty lines

File: TESTS/spm/neg_server_tests/neg_ipc_tests.cpp [hunk 4]
```diff
@@ -235,6 +249,26 @@ void server_end_invalid_handle()
...
+    psa_connect(PART1_END_INVALID_RETVAL_CONNECT, MINOR_VER);
+   
+    TEST_FAIL_MESSAGE("server_end_invalid_retval_connect negative test failed at client side");
+}
+
+//Testing server end invalid retval for call
+void server_end_invalid_retval_call()
+{
+    psa_handle_t handle = PSA_NULL_HANDLE;
+
```
Comment (orenc17) [inline TESTS/spm/neg_server_tests/neg_ipc_tests.cpp]: empty lines

File: TESTS/spm/neg_server_tests/neg_ipc_tests.cpp [hunk 5]
```diff
@@ -235,6 +249,26 @@ void server_end_invalid_handle()
     TEST_FAIL_MESSAGE("server_end_invalid_handle negative test failed at client side");
 }
 
+//Testing server end invalid retval for connect
+void server_end_invalid_retval_connect()
+{
+    psa_connect(PART1_END_INVALID_RETVAL_CONNECT, MINOR_VER);
+
```
Comment (orenc17) [inline TESTS/spm/neg_server_tests/neg_ipc_tests.cpp]: empty lines

File: TESTS/spm/neg_client_tests/neg_ipc_tests.cpp [hunk 6]
```diff
@@ -104,14 +104,24 @@ void client_connect_invalid_pol_ver_RELAXED()
...
                );
 
-    TEST_FAIL_MESSAGE("client_connect_invalid_pol_ver_STRICT negative test failed");
+    TEST_FAIL_MESSAGE("client_connect_invalid_high_pol_ver_STRICT negative test failed");
+}
+
+//Testing client connect version policy is STRICT and requeted version equales the minimum version
+void client_connect_invalid_equal_pol_ver_STRICT()
+{
+    psa_connect( NEG_CLIENT_PART1_ROT_SRV2,           //NEG_CLIENT_PART1_ROT_SRV2 should have STRICT policy
```
Comment (orenc17) [inline TESTS/spm/neg_client_tests/neg_ipc_tests.cpp]: one line, comment above it

File: spm/spm_server.c (hunk 1 above)
Comment (danny4478) [reply spm/spm_server.c]: * Why do you think this is panic?
This is a coding error which can cause this to fail, not an outside attack. 
* Changed condition as you requested

File: TESTS/spm/neg_server_tests/neg_ipc_tests.cpp (hunk 2 above)
Comment (danny4478) [reply TESTS/spm/neg_server_tests/neg_ipc_tests.cpp]: Done

File: TESTS/spm/neg_server_tests/neg_ipc_tests.cpp (hunk 3 above)
Comment (danny4478) [reply TESTS/spm/neg_server_tests/neg_ipc_tests.cpp]: Done

**Comment by orenc17** [comment]
:lgtm:

---

Reviewed 4 of 8 files at r1, 2 of 2 files at r2, 2 of 2 files at r3.
Review status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @alzix, @mikisch81, and @romkuz01)

---



*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/44#-:-LG0Cl1T0fg_MvfkIFH1:bnfp4nl)*

**Comment by orenc17** [comment]
@danny4478 Great job you A**-hole

41615731-b5d1dd24-7404-11e8-83fe-fcbb40fd283b

**Pull request merged**